### ISSUE TITLE
PROXY: Can't use hostname for Srv_Addr

### ISSUE THREAD

**Issue by figassis** [nfs-ganesha/nfs-ganesha#524]
I have the configuration below, however, which keeps giving the error:

(/etc/ganesha/ganesha.conf:12): Expected an IP address, got a option name or number

I'm proxying Amazon EFS, and it provides a DNS name to resolve to the closest AZ, so using an IP defeats the purpose.

Can anyone help?

```
EXPORT {
	Export_Id = 0;
	Path = /export;
	Pseudo = /;
	Access_Type = RW;
	Squash = No_Root_Squash;
	Protocols = 4;
	Transports = "TCP";
	SecType = "sys";
	FSAL {
		Name = PROXY;
        Srv_Addr = my.valid.servername;
        Use_Privileged_Client_Port = true;
	}
}

NFSV4 {
    GRACELESS = true;
}
```

**Comment by malahal** [comment]
Looking at the code, it is hard coded to have IP address (not a host name).

**Comment by ffilz** [comment]
Let's put this one in the backlog.

**Comment by boulos** [comment]
There isn't currently a "hostname:port" in config, right? As malahal pointed out, srvaddr in the proxy code is CONF_MAND_IP_ADDR / CONF_ITEM_IP_ADDR .

FSAL_GLUSTER seems to do:

export.c:	CONF_MAND_STR("hostname", 1, MAXPATHLEN, NULL,

A challenge with hostnames though is to determine the semantics of resolution. Should code resolve the hostname to an IP "on boot" (in this case, on mount), periodically (to allow fail over / load balancing), for *every* op (unlikely), or something else?

It seems like a fine alternative to allow hostname (with resolution on "boot"), though I think it would be best if Ganesha had a config for "valid hostname" following https://tools.ietf.org/html/rfc1035#section-2.3.1 and then a "hostname or IP" config item.

**Comment by boulos** [comment]
I don't think this is necessarily specific to `PROXY` cases. We should probably instead hav the ability for the config system to support hostnames directly (e.g., this presumably applies to Gluster, Ceph, etc.).
